proxy/ProxyClientTransaction.h

```diff
@@ -211,7 +211,11 @@ class ProxyClientTransaction : public VConnection
 
   virtual bool allow_half_open() const = 0;
 
-  virtual const char *get_protocol_string() const = 0;
+  virtual const char *
```

Comment:
Currently plugins can not set client_protocol of HttpSM ( which is going to be added by this patch ), but it could be set in the future. And it also has consistency for other stats in HttpSM (e.g. client_sec_protocol, client_cipher_suite). So, let's leave it string.

I'll file a new tickets for optimization.